avm/ptn/app-service-lza/hosting-environment/modules/app-service/app-service.module.bicep

```
@@ -0,0 +1,336 @@
+@description('Optional, default is false. Set to true if you want to deploy ASE v3 instead of Multitenant App Service Plan.')
+param deployAseV3 bool = false
+
+@description('Optional if deployAseV3 = false. The identifier for the App Service Environment v3 resource.')
+@minLength(1)
+@maxLength(36)
+param aseName string
+
+@description('Required. Name of the App Service Plan.')
+@minLength(1)
+@maxLength(40)
+param appServicePlanName string
+
+@description('Required. Name of the web app.')
+@maxLength(60)
+param webAppName string
+
+@description('Required. Name of the managed Identity that will be assigned to the web app.')
+@minLength(3)
+@maxLength(128)
+param managedIdentityName string
+
+@description('Optional S1 is default. Defines the name, tier, size, family and capacity of the App Service Plan. Plans ending to _AZ, are deploying at least three instances in three Availability Zones. EP* is only for functions')
+@allowed([
+  'S1'
+  'S2'
+  'S3'
+  'P1V3'
+  'P2V3'
+  'P3V3'
+  'EP1'
+  'EP2'
+  'EP3'
+  'ASE_I1V2'
+  'ASE_I2V2'
+  'ASE_I3V2'
+])
+param sku string
+
+@description('Optional. Set to true if you want to deploy the App Service Plan in a zone redundant manner. Default is true.')
+param zoneRedundant bool = true
+
+@description('Optional. Location for all resources.')
+param location string
+
+@description('Resource tags that we might need to add to all resources (i.e. Environment, Cost center, application name etc)')
+param tags object
+
+@description('Default is empty. If empty no Private Endpoint will be created for the resoure. Otherwise, the subnet where the private endpoint will be attached to')
+param subnetPrivateEndpointId string = ''
```

```suggestion
param subnetPrivateEndpointResourceId string = ''
```
as per [SNFR22](https://azure.github.io/Azure-Verified-Modules/specs/bcp/res/#:~:text=ID%3A%20SNFR22%20%2D%20Category%3A%20Inputs%20%2D%20Parameters/Variables%20for%20Resource%20IDs)